packages/aws-cdk-lib/core/test/removal-policies.test.ts

```diff
@@ -0,0 +1,252 @@
+    const resource = new TestResource(parent, 'Resource');
+
+    // WHEN
+    RemovalPolicies.of(parent).retain({
+      applyToResourceTypes: [
+        TestBucketResource.CFN_RESOURCE_TYPE_NAME, // 'AWS::S3::Bucket'
+        table.cfnResourceType, // 'AWS::DynamoDB::Table'
+      ],
+    });
+
+    // THEN
+    synthesize(stack);
+    expect(bucket.cfnOptions.deletionPolicy).toBe('Retain');
+    expect(table.cfnOptions.deletionPolicy).toBe('Retain');
+    expect(resource.cfnOptions.deletionPolicy).toBeUndefined();
+  });
+
+  test('excludes specified resource types using strings', () => {
+    // GIVEN
+    const stack = new Stack();
+    const parent = new Construct(stack, 'Parent');
+    const bucket = new TestBucketResource(parent, 'Bucket');
+    const table = new TestTableResource(parent, 'Table');
+    const resource = new TestResource(parent, 'Resource');
+
+    // WHEN
+    RemovalPolicies.of(parent).snapshot({
+      excludeResourceTypes: [
+        TestResource.CFN_RESOURCE_TYPE_NAME, // 'AWS::Test::Resource'
+      ],
+    });
+
+    // THEN
+    synthesize(stack);
+    expect(bucket.cfnOptions.deletionPolicy).toBe('Snapshot');
+    expect(table.cfnOptions.deletionPolicy).toBe('Snapshot');
+    expect(resource.cfnOptions.deletionPolicy).toBeUndefined();
+  });
+
+  test('excludes specified resource types using classes', () => {
+    // GIVEN
+    const stack = new Stack();
+    const parent = new Construct(stack, 'Parent');
+    const bucket = new TestBucketResource(parent, 'Bucket');
+    const table = new TestTableResource(parent, 'Table');
+    const resource = new TestResource(parent, 'Resource');
+
+    // WHEN
+    RemovalPolicies.of(parent).snapshot({
+      excludeResourceTypes: [
+        resource.cfnResourceType,
+      ],
+    });
+
+    // THEN
+    synthesize(stack);
+    expect(bucket.cfnOptions.deletionPolicy).toBe('Snapshot');
+    expect(table.cfnOptions.deletionPolicy).toBe('Snapshot');
+    expect(resource.cfnOptions.deletionPolicy).toBeUndefined();
+  });
+
+  test('applies different removal policies', () => {
+    // GIVEN
+    const stack = new Stack();
+    const destroy = new TestResource(stack, 'DestroyResource');
+    const retain = new TestResource(stack, 'RetainResource');
+    const snapshot = new TestResource(stack, 'SnapshotResource');
+    const retainOnUpdate = new TestResource(stack, 'RetainOnUpdateResource');
+
+    // WHEN
+    RemovalPolicies.of(destroy).destroy();
+    RemovalPolicies.of(retain).retain();
+    RemovalPolicies.of(snapshot).snapshot();
+    RemovalPolicies.of(retainOnUpdate).retainOnUpdateOrDelete();
+
+    // THEN
+    synthesize(stack);
+    expect(destroy.cfnOptions.deletionPolicy).toBe('Delete');
+    expect(retain.cfnOptions.deletionPolicy).toBe('Retain');
+    expect(snapshot.cfnOptions.deletionPolicy).toBe('Snapshot');
+    expect(retainOnUpdate.cfnOptions.deletionPolicy).toBe('RetainExceptOnCreate');
+  });
+
+  test('last applied removal policy takes precedence', () => {
```

Comment:
I think the following explanation is correct.

```suggestion
  test('last applied removal policy takes precedence only if `overwrite` is set to true', () => {
```